k6ext/k6test/vu.go

```diff
@@ -57,10 +57,11 @@ func (v *VU) AssertSamples(assertSample func(s k6metrics.Sample)) int {
 // so that the test can read the metrics being emitted to the channel.
 type WithSamplesListener chan k6metrics.SampleContainer
 
-// WithLookupFunc a custom lookup function that returns test values.
-type WithLookupFunc env.LookupFunc
-
```

Comment:
Why did we remove this and now just accept a "direct" `env.LookupFunc`? Isn't it more idiomatic to keep it this way? Did we remove it because it was making declaration too long? E.g: `withLookupFunc(env.ConstLookup(env.BrowserHeadless, "0"))` ?

Reply:
Idiomatic Go is about being concise. From the Effective Go: _"A helpful doc comment can often be more valuable than an extra long name."_.

`WithLookupFunc` makes the usage unnecessarily longer and causes unnecessary roundtrips between types (`WithLookupFunc` to `env.LookupFunc`). `NewVU` checks the options by their types, and there is an already `env.LookupFunc` type. Instead of using `withLookupFunc` everywhere, I used the documentation to make it clear where to use what and how.

https://github.com/grafana/xk6-browser/blob/e5fbe425bbb3a2c918a5f29c133e350045a4b951/k6ext/k6test/vu.go#L60-L64